helper/stats/cpu.go

```diff
@@ -19,63 +19,98 @@ const (
 )
 
 var (
-	cpuMhzPerCore float64
-	cpuModelName  string
-	cpuNumCores   int
-	cpuTotalTicks float64
+	cpuPowerCoreCount      int
+	cpuPowerCoreMHz        uint64
+	cpuEfficiencyCoreCount int
+	cpuEfficiencyCoreMHz   uint64
+	cpuTotalTicks          uint64
+	cpuModelName           string
+)
 
+var (
 	initErr error
 	onceLer sync.Once
 )
 
 func Init() error {
 	onceLer.Do(func() {
-		var merrs *multierror.Error
-		var err error
-		if cpuNumCores, err = cpu.Counts(true); err != nil {
-			merrs = multierror.Append(merrs, fmt.Errorf("Unable to determine the number of CPU cores available: %v", err))
-		}
+		switch {
+		case m1cpu.IsAppleSilicon():
+			cpuModelName = m1cpu.ModelName()
+			cpuPowerCoreCount = m1cpu.PCoreCount()
+			cpuPowerCoreMHz = m1cpu.PCoreHz() / 1_000_000
+			cpuEfficiencyCoreCount = m1cpu.ECoreCount()
+			cpuEfficiencyCoreMHz = m1cpu.ECoreHz() / 1_000_000
+			bigTicks := uint64(cpuPowerCoreCount) * cpuPowerCoreMHz
+			littleTicks := uint64(cpuEfficiencyCoreCount) * cpuEfficiencyCoreMHz
+			cpuTotalTicks = bigTicks + littleTicks
+		default:
+			// for now, all other cpu types assume only power cores
+			// todo: this is already not true for Intel 13th generation
 
-		var cpuInfo []cpu.InfoStat
-		ctx, cancel := context.WithTimeout(context.Background(), cpuInfoTimeout)
-		defer cancel()
-		if cpuInfo, err = cpu.InfoWithContext(ctx); err != nil {
-			merrs = multierror.Append(merrs, fmt.Errorf("Unable to obtain CPU information: %v", err))
-		}
+			var err error
+			if cpuPowerCoreCount, err = cpu.Counts(true); err != nil {
+				initErr = errors.Join(initErr, fmt.Errorf("failed to detect number of CPU cores: %v", err))
+			}
 
-		for _, cpu := range cpuInfo {
-			cpuModelName = cpu.ModelName
-			cpuMhzPerCore = cpu.Mhz
-			break
-		}
+			ctx, cancel := context.WithTimeout(context.Background(), cpuInfoTimeout)
+			defer cancel()
+
+			var cpuInfo []cpu.InfoStat
+			if cpuInfo, err = cpu.InfoWithContext(ctx); err != nil {
+				errors.Join(err, fmt.Errorf("failed to detect CPU information: %v", err))
```

Comment:
> // Join returns an error that wraps the given errors.

So, I think we need to capture the returned error.

Reply:
yeah @jrasell is right - this line actually does nothing because the result is returned from `Join` ... however this line is unnecessary anyway since the assignment to initErr already contains a contextual message